hw2/04-background.js

```diff
@@ -1 +1,59 @@
 // Add your code here
+const START_CLASS = "btn-primary";
+const STOP_CLASS = "btn-danger";
+const BASE_TRANSITION_DURATION = 3000;
+const BASE_BACKGROUND_CLASS = "bg-primary";
+
+const input = document.querySelector("#interval");
+const callToActionButton = document.querySelector("#cta");
+const body = document.querySelector("body");
+body.style.transitionDuration = `${BASE_TRANSITION_DURATION}ms`;
+
+let interval;
+
+const startInterval = () => {
+  body.classList.remove(BASE_BACKGROUND_CLASS);
+  const newIntervalInMs = input.value * 1000;
+
+  interval = setInterval(() => {
+    const newRGBA = `rgba(${getNewRandomColorValue()}, ${getNewRandomColorValue()}, ${getNewRandomColorValue()}, 0.5)`;
```

Comment:
Instead of calling the getNewRandomColorValue() three times on the same line, it would probably be more efficient to refactor the code so that the function is called once, but have it return 3 color values.

Reply:
can you give me a code snippet on how to do that ❓ wouldn't that make us do `Math.floor(Math.random() * 256)` 3 times ❓

Reply:
```suggestion
    const {color1, color2, color3} = returnedObjectOfThreeRandomColors
    const newRGBA = `rgba(${color1}, ${color2}, ${color1}, 0.5)`;
```

Reply:
yes I understand but then how would you implement `returnedObjectOfThreeRandomColors` ❓

Reply:
returnedObjectOfThreeRandomColors would be the return value of calling a function that returns an object of three random colors. 

So instead of this 

```
const getNewRandomColorValue = () => Math.floor(Math.random() * 256);

```

You would need something like this


```
     const getNewRandomColorValues = () => {
     
    //generate 3 random colors here however you'd like
    
    //then just return them all at once in an object
    
    return { color1: colorValue1, color2: colorValue2, color3: colorValue3}
    
     }
    
```

Reply:
right, but then we'll have to repeat `Math.floor(Math.random() * 256)` 3 times, which ... is not that efficient based on your initial comment ❓ Also, wouldn't repeating `Math.floor(Math.random() * 256)` 3 times make it less readable than repeating `getNewRandomColorValues` 3 times❓

Reply:
I really like your comments, this is good dialogue. I'll definitely be looking into this more. As for your comment on whether the latter is more readable, your opinion is just as valid now that you explain your reasoning, I think at the end of the day this could boil down to personal preference. I particularly mentioned the object destructing technique that I did because you will see similar things in React.

Reply:
Yep! I really appreciate your thoughts too! 

I did actually think about that and one solution I came up with is to do something like [this](https://www.typescriptlang.org/play?ts=4.2.3#code/MYewdgzgLgBA5gUygOQQdwEoEMwBMQC2AwiADYgBOAalqQK4IwC8MAFAJTMB8MA3gLAAoGDFCRYochQCCFClgCezGAG0AugG4hIgGaU2pJDACWygAwaTMADwwAzJeMBqJ5wHCRnyZVnyFAOgAHOggAC1YAWSwoUP8dckpI6Nj5PEIOGAAqGAAmAFYANnZ2LQ8AXyFtGAokOgowGAADCjgAIyxWABJebxk5RX8AKxBjMFYAIgAaGHH2Mumzfzz2RtKKwSExCDIEf3I4VkQUdGw04jJKGnoEDhKgA)
but... that was too overkill for a single use case like this so I decided to just stick with the 
`const getNewRandomColorValue = () => Math.floor(Math.random() * 256);`

Reply:
For now because the scope of this assignment is small and the actual computation done by the getNewRandomColorValue() is minimal I don't think it would make too much of a difference. However, if getNewRandomColorValue() was some computationally heavy function, I couldn't imaging calling it three times within a template string would be an efficient approach. I need to find a study or data on this to see if my theory is accurate, for now just take it as my personal preference, which you don't need to follow. 

All in all, great good on this assignment. Keep up the good work, I will be looking forward to checking out your HW3!

Reply:
I... already merge it so I won't change it again anyway, but I totally agree with your preference though

I'll do it once I we have a bigger scope!